### PR TITLE
Added Merge Sort Algorithm

### PR DESCRIPTION
**Pull request by codersanjeev**

### CODE REVIEW

**Comment by codersanjeev** [comment]
@vbrazo, please review

File: sorts/MergeSort.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	
+	mutating private func _merge(
+		from left: Int,
+		mid: Int,
+		to right: Int,
+		by comparison: (Element, Element) -> Bool
+	) {
+		var copy = [Element](repeating: self[left], count: right - left + 1)
+		var (p, q, k) = (left, mid + 1, 0)
```
Comment (TiagoMaiaL) [inline sorts/MergeSort.swift:32]: What does `p, q, k` stand for?

File: sorts/MergeSort.swift [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+				copy[k] = self[p]
+				k += 1
+				p += 1
+			} else if comparison(self[p], self[q]) {
+				copy[k] = self[p]
+				k += 1
+				p += 1
+			} else {
+				copy[k] = self[q]
+				k += 1
```
Comment (TiagoMaiaL) [inline sorts/MergeSort.swift:48]: `k += 1` is getting incremented in each branch. All four calls could be moved to the end of the `for` loop.

Comment (codersanjeev) [reply sorts/MergeSort.swift]: done

File: sorts/MergeSort.swift (hunk 1 above)
Comment (codersanjeev) [reply sorts/MergeSort.swift]: done, changed variable names to more descriptive names and also, added some visual representation in comments.

File: sorts/MergeSort.swift [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	func mergeSorted(by comparison: (Element, Element) -> Bool) -> Array {
+		var copy = self
+		copy.mergeSort(by: comparison)
+		return copy
+	}
+	
+}
+
+// The code below can be used for testing
+var numberList = [15, 2, 23, 11, 3, 9]
```
Comment (TiagoMaiaL) [inline sorts/MergeSort.swift:66]: When trying to compile this file, Xcode gives the following errors:
Screen Shot 2021-07-13 at 12 33 25

I suggest commenting the test code, so anyone interested in this file is able to compile it.

Comment (codersanjeev) [reply sorts/MergeSort.swift]: done

**Comment by TiagoMaiaL** [comment]
Looks good to me 👍.

**Review by TiagoMaiaL — APPROVED** [review]

**Comment by codersanjeev** [comment]
Can this be merged now? @TiagoMaiaL  @anantcodes @carloshmartins

**Comment by TiagoMaiaL** [comment]
> Can this be merged now? @TiagoMaiaL @anantcodes @carloshmartins

I don't have the permissions. @carloshmartins merged the last one I opened.

**Comment by carloshmartins** [comment]
LGTM

**Pull request merged**